cpp/modmesh/toggle/RadixTree.cpp

```diff
@@ -72,6 +72,132 @@ void CallProfiler::print_profiling_result(const RadixTreeNode<CallerProfile> & n
     }
 }
 
+void CallProfilerSerializer::serialize_call_profiler(const CallProfiler & profiler, std::ostream & outstream)
+{
+    // Example:
+    //  {
+    //      "radix_tree":
+    //      {
+    //          "nodes":[
+    //              {"key":-1,"name":"","data":{"start_time": 0,"caller_name": "","total_time": 0,"call_count": 0,"is_running": 0},"children":[0]},
+    //              {"key":0,"name":"void modmesh::detail::foo1()","data":{"start_time": 17745276708555250,"caller_name": "void modmesh::detail::foo1()","total_time": 61002916,"call_count": 1,"is_running": 1},"children":[1]},
+    //              {"key":1,"name":"void modmesh::detail::foo2()","data":{"start_time": 17745276708555458,"caller_name": "void modmesh::detail::foo2()","total_time": 54002250,"call_count": 1,"is_running": 1},"children":[2]},
+    //              {"key":2,"name":"void modmesh::detail::foo3()","data":{"start_time": 17745276743555833,"caller_name": "void modmesh::detail::foo3()","total_time": 19001833,"call_count": 1,"is_running": 1},"children":[]}
+    //          ],
+    //          "current_node":-1,
+    //          "unique_id":3
+    //      }
+    //  }
```

Comment:
Additionally, the dictionary, that map the function to keyID, should be added into the radix tree.

Reply:
By the address of the node in the nodes, do you mean the address of the stack frame?

Reply:
> By the address of the node in the nodes, do you mean the address of the stack frame?
> 
> Maybe it is worth a try, but I am not 100% sure the stack frame address can be used to identify a node.

Yes, it is.
I just record the relationship between nodes, 
and I actually don't want to get the information from the real address.

By the way, I try to casting the pointer to the unsigned long when serializing.
It seems that we couldn't directly print the address in ubuntuOS.

Reply:
> > By the address of the node in the nodes, do you mean the address of the stack frame?
> > Maybe it is worth a try, but I am not 100% sure the stack frame address can be used to identify a node.
> 
> Yes, it is. I just record the relationship between nodes, and I actually don't want to get the information from the real address.
> 
> By the way, I try to casting the pointer to the unsigned long when serializing. It seems that we couldn't directly print the address in ubuntuOS.

You are right. Here is a bug.
https://github.com/solvcon/modmesh/actions/runs/9338060610/job/25700741584?pr=323
It seems that we couldn't identify a node by stack frame address because of the forbidden of the reinterpret_cast.
I think I should number the node by myself when Breadth-first searching.